Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py

```diff
@@ -0,0 +1,137 @@
+import cv2   # noqa: F401
+import numpy as np
+from PIL import Image
+import io
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+# sharpened
+def sharpened(image: np.ndarray):
+    """
+    Sharpens image according to selected values.
+    Args:
+        image(np.ndarray): the image that would be sharpened.
+    Returns:
+        (np.ndarray).
+    """
+    kernel = np.array([[-1, -1, -1], [-1, 8, -1], [-1, -1, 0]], np.float32)
+    kernel = 1 / 3 * kernel
+    sharp = cv2.filter2D(image, -1, kernel)  # pylint: disable=E1101
+    return sharp
+
+
+def image_resize(image: Image, width: int, height: int):
+    """
+    REsizes image according to width and heights values.
+    Args:
+        image(np.ndarray): the image that would be sharpened.
+        width(int): new width.
+        height(int): new height.
+    Returns:
+        (Image).
+    """
+    image = image.resize((width, height), Image.ANTIALIAS)
+    return image
+
+
+def action_grey() -> dict[str, Any]:
+    """
+    Generate grayscale image.
+    Returns:
+        FileResult (dict[str, Any]).
+    """
+    args = demisto.args()
+    entry_id = args.get('file_entry_id')
+    width = arg_to_number(args.get('image_resize_width'))
+    height = arg_to_number(args.get('image_resize_height'))
+    image, name = get_file_details(entry_id)
+    stream_gray = io.BytesIO()
```

Comment:
It seems that you're using the same code over and over in all functions. 
Please move it to a helper function and pass to it the relevant stuff for each command